dulwich/porcelain.py

```diff
@@ -2245,13 +2248,13 @@ def describe(repo, abbrev=7):
                         return "{}-{}-g{}".format(
                             tag_name,
                             commit_count,
-                            latest_commit.id.decode("ascii")[:abbrev],
+                            latest_commit.id.decode("ascii")[:abbrev or 7],
```

Comment:
Strictly speaking I suppose this would affect `abbrev=0` callsites; I could adapt the code to use a `kwargs` pattern if it'd be preferable to reduce the behavioural diff.

Reply:
(or check if `is not None` in a ternary)